x-pack/plugins/alerting/server/saved_objects/rules_settings_mappings.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsTypeMappingDefinition } from '@kbn/core/server';
+
+export const rulesSettingsMappings: SavedObjectsTypeMappingDefinition = {
+  properties: {
+    flapping: {
+      properties: {
+        enabled: {
+          type: 'boolean',
+        },
+        lookBackWindow: {
+          type: 'long',
+        },
+        statusChangeThreshold: {
+          type: 'long',
+        },
+        createdBy: {
+          type: 'keyword',
+        },
+        updatedBy: {
+          type: 'keyword',
+        },
+        createdAt: {
+          type: 'date',
+        },
+        updatedAt: {
+          type: 'date',
+        },
+      },
+    },
+  },
+};
```

Comment:
Q: Do we need all these fields indexed, or can we add `index: false` to any of those?

Reply:
I think at the very least we can add `index: false` to the metadata fields (createdBy, etc). Since these settings saved objects are instantiated once per space, we are only querying by the ID anyways. 

@mikecote @XavierM curious to hear if there's any reason to index the metadata fields?

Reply:
yes I agree with @JiaweiWu